src/libcore/macros.rs

```diff
@@ -542,7 +542,9 @@ macro_rules! unreachable {
 #[macro_export]
 #[stable(feature = "rust1", since = "1.0.0")]
 macro_rules! unimplemented {
-    () => (panic!("not yet implemented"))
+    () => (panic!("not yet implemented"));
+    ($msg:expr) => (unimplemented!("{}", $msg));
```

Comment:
This like could be changed to this: `panic!(format_args!(concat!("not yet implemented: ", $msg)))`, and it would catch `unimplemented!("foo {}")`.  The error message it prints:

```
error: invalid reference to argument `0` (no arguments given)
  --> <anon>:6:29
   |
6  |         panic!(format_args!(concat!("not yet implemented: ", $msg)))
   |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
13 |     unimplemented!("foo {}")
   |     ----------------- in this macro invocation
```

It's not super clear, but it would catch that mistake. However, it differs from `unreachable`, and even `panic`.